karapuzio/assignment4/Island.java

```diff
@@ -0,0 +1,97 @@
+package edu.codeU.assignment4;
+
+public class Island {
+
+    /**
+     * Number of rows in map.
+     */
+    private int rows;
+    /**
+     * Number of columns in map.
+     */
+    private int columns;
+    /**
+     * The map where we look for islands.
+     */
+    private boolean[][] map;
+    /**
+     * Flag array that is used in traversal map using queue structure.
+     */
+    private boolean[][] flag;
```

Comment:
You can simplify your algorithm a lot. Can you think of solving it without this additional grid of booleans? Let me know if you need any additional pointers. But try thinking how you can solve it using just the grid that was originally provided.